### PR TITLE
Only require 'distro' on linux (as it is linux specific)

### PR DESCRIPTION
**Pull request by grembo**
This is a trivial/obvious fix.

### CODE REVIEW

File: master/setup.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ def define_plugin_entries(groups):
     'txaio ' + txaio_ver,
     'autobahn ' + autobahn_ver,
     'PyJWT',
-    'distro'
+    'distro;platform_system==="Linux"'
```
Comment (tardyp) [inline master/setup.py]: I wasn't aware of this construct. Is this supported by all pip/setuptools versions?
Does that work with wheels?

Comment (grembo) [inline master/setup.py]: This was defined in https://www.python.org/dev/peps/pep-0508/. Based on this post https://discourse.numenta.org/t/setup-py-error-invalid-environment-marker/1298/3 it's supported by:

 - setuptools: 20.2.2 +
 - wheel: 0.24.0 +
 - pip: 8.1.2 +

Comment (seankelly) [inline master/setup.py]: That is a much newer version of setuptools than currently required.

Comment (grembo) [inline master/setup.py]: If this is critical, the alternative would be something like

```
import platform
...
if platform.system() == 'Linux':
    setup_args['install_requires'].append('distro')

```
If you prefer this, I can update the pull request.

Comment (grembo) [inline master/setup.py]: updated to use a more conservative approach to accomplish the same

Comment (tardyp) [inline master/setup.py]: Problem with that approach is that it does not work with wheels. As I generate the universal wheels on my mac, it wont include distro anymore in the wheels deps.

I would be more in favor of the combination, which would test setup tools version :-/

Comment (grembo) [inline master/setup.py]: Can you give me an example of what your suggested approach would look like?

Comment (tardyp) [inline master/setup.py]:
```python
import platform
from distutils.version import LooseVersion
import setuptools
...
if LooseVersion(setuptools.version.__version__) >= LooseVersion("20.2.2"):
    setup_args['install_requires'].append('distro;platform_system==="Linux')
elif platform.system() == 'Linux':
    setup_args['install_requires'].append('distro')
```

**Comment by tardyp** [comment]
@grembo Hope you don't mind, in order to move faster I did edit your PR last commit to include the proposed change. Let me know if this works for you (and your BSD port)

**Comment by grembo** [comment]
@tardyp Sorry for the late response, this is ok with me. I'll change/remove the patch to the port on the next release, so it's all good.

**Pull request merged**